### DISCUSSION TITLE
Systray widget crash message when on dual screen or more screens

### DISCUSSION THREAD

**Discussion by erikdubois** [qtile/qtile#5040] [Ideas]
Systray widget can only be there once ... hence on the second screen the systray is replaces with this message

"Widget crashed; Systray (click to hide)"

If you click it, it is indeed gone.

Can we get a variable to hide this message?

**Comment by elParaguayo** [comment]
Well, it's definitely possible but I don't really think we want to be hiding crash messages as they're useful for users to know when there's a problem with their setup.

Is there any reason why you can't just have one Systray widget in your config?

We could automatically remove the second instance of the widget like we do with widgets that are only compatible with either the x11 or Wayland backends.

**Comment by erikdubois** [comment]
I would like to remove the second instance of that widget
but not automatically
leave it up to user to remove a second instance on a per widget basis
when you think about it ... why would you not allow a second systray on your dual screen monitor 2?

![ArcoLinux-2024-10-18-1729232028_screenshot_3840x1080](https://github.com/user-attachments/assets/4a0f97a6-c1c2-4259-b54b-5c0347da8d88)

See second screen - top right - it is just not nice

**Reply by elParaguayo** [reply]
It's not that we don't want to allow it. The problem is with how Systray works on x11. It's a window where app icons are also windows that are reparented into the widget. This means we can't duplicate the contents in multiple windows. 

Even if we could copy the icon's appearance, this wouldn't copy the context menu to the other screen.

We could add a `hide_crash` parameter to all widgets which could be set on a per widget basis, or globally by setting it in `widget_defaults`.

**Comment by erikdubois** [comment]
sounds interesting... if you can put it on the to-do list

**Reply by elParaguayo** [reply]
#5045 should address this.

**Reply by elParaguayo** [reply]
This has been merged - but isn't in 0.29.0 release.
